### PR TITLE
document upload error reworked

### PR DESCRIPTION
**Pull request by OgunyemiO**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EXUI-1453

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts [hunk 1]
```diff
@@ -280,25 +280,30 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
...
-      if (start >= 0) {
-        const json = fullError.substring(start, fullError.length - 1).split('<EOL>').join('');
-        const obj = JSON.parse(json);
-        if (obj?.error) {
-          errorMsg = obj.error;
+    switch (error.status) {
+      case 0:
+        return WriteDocumentFieldComponent.UPLOAD_ERROR_NOT_AVAILABLE;
+      case 502:
+        return WriteDocumentFieldComponent.UPLOAD_ERROR_NOT_AVAILABLE;
```
Comment (RiteshHMCTS) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts:287]: @OgunyemiO the above two switch case you can merge together and return single return statement like below
case 0:
case 502:
return WriteDocumentFieldComponent.UPLOAD_ERROR_NOT_AVAILABLE;

**Review by andywilkinshmcts — APPROVED** [review]
Thanks Ola

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts (hunk 1 above)
Comment (OgunyemiO) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts:286]: Done!! @RiteshHMCTS

**Review by RiteshHMCTS — APPROVED** [review]

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts [hunk 2]
```diff
@@ -280,25 +280,29 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
...
+            if (obj?.error) {
+              errorMsg = obj.error;
+            }
+          }
         }
-
-      }
+        return errorMsg;
+      case 429:
+        return error.message;
```
Comment (olusegz07) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts:302]: Please change error.message to error?.error

Comment (OgunyemiO) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts]: Done @olusegz07

**Review by RiteshHMCTS — APPROVED** [review]

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts [hunk 3]
```diff
@@ -280,25 +280,29 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
...
+            }
+          }
         }
-
-      }
+        return errorMsg;
+      case 429:
+        return error.error;
+      default:
+        return "Error Uploading File";
```
Comment (andywilkinshmcts) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts:304]: Use a constant for this

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts [hunk 4]
```diff
@@ -280,25 +280,29 @@ export class WriteDocumentFieldComponent extends AbstractFieldWriteComponent imp
...
-        const json = fullError.substring(start, fullError.length - 1).split('<EOL>').join('');
-        const obj = JSON.parse(json);
-        if (obj?.error) {
-          errorMsg = obj.error;
+    switch (error.status) {
+      case 0:
+      case 502:
+        return WriteDocumentFieldComponent.UPLOAD_ERROR_NOT_AVAILABLE;
+      case 422:
+        let errorMsg = 'Error uploading file';
```
Comment (andywilkinshmcts) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts:288]: DRY - 'Error uploading file' is in twice, though with different capitalisation. Please add a constant for this.

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts (hunk 4 above)
Comment (OgunyemiO) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts]: @andywilkinshmcts Done!

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts (hunk 3 above)
Comment (OgunyemiO) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/document/write-document-field.component.ts]: @andywilkinshmcts Done!

**Review by andywilkinshmcts — APPROVED** [review]

**Review by olusegz07 — APPROVED** [review]

**Review by andywilkinshmcts — APPROVED** [review]

**Pull request merged**